pkg/logs/launchers/integration/launcher.go

```diff
@@ -156,58 +281,104 @@ func (s *Launcher) makeFileSource(source *sources.LogSource, logFilePath string)
+	filepath := s.integrationLogFilePath(source)
 
-	file, err := os.Create(logFilePath)
+	file, err := os.Create(filepath)
 	if err != nil {
-		return "", nil
+		ddLog.Error("Error creating file for log source:", err)
+		return nil, err
 	}
-	defer file.Close()
+	ddLog.Info("Successfully created integrations log file:", file.Name())
 
-	return logFilePath, nil
+	err = file.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	fileInfo := &FileInfo{
+		filename:     filepath,
+		lastModified: time.Now(),
+		size:         0,
+	}
+
+	return fileInfo, nil
 }
 
 // integrationLoglogFilePath returns a file path to use for an integration log file
 func (s *Launcher) integrationLogFilePath(id string) string {
-	fileName := strings.ReplaceAll(id, " ", "-")
-	fileName = strings.ReplaceAll(fileName, ":", "_") + ".log"
+	fileName := strings.ReplaceAll(id, ":", "_") + ".log"
 	logFilePath := filepath.Join(s.runPath, fileName)
 
 	return logFilePath
 }
 
-// ensureFileSize enforces the max file size for files integrations logs
-// files. Files over the set size will be deleted and remade.
-func (s *Launcher) ensureFileSize(logFilePath string) error {
-	maxFileSizeSetting := pkgconfigsetup.Datadog().GetInt64("logs_config.integrations_logs_files_max_size")
-	maxFileSizeBytes := maxFileSizeSetting * 1024 * 1024
-
-	fi, err := os.Stat(logFilePath)
+// computerDiskUsageMax computes the max disk space the launcher can use based
+// off the integrations_logs_disk_ratio and integrations_logs_total_usage
+// settings
+func computeMaxDiskUsage(runPath string, logsTotalUsageSetting int64, usageRatio float64) (int64, error) {
+	usage, err := filesystem.NewDisk().GetUsage(runPath)
 	if err != nil {
-		return err
+		return 0, err
 	}
 
-	if fi.Size() > int64(maxFileSizeBytes) {
-		err := os.Remove(logFilePath)
+	diskReserved := float64(usage.Total) * (1 - usageRatio)
+	diskAvailable := int64(usage.Available) - int64(math.Ceil(diskReserved))
+
+	return min(logsTotalUsageSetting, diskAvailable), nil
+}
+
+// scanInitialFiles scans the run path for initial files and then adds them to
+// be managed by the launcher
+func (s *Launcher) scanInitialFiles(dir string) error {
+	err := filepath.Walk(dir, func(_ string, info os.FileInfo, err error) error {
 		if err != nil {
-			if os.IsNotExist(err) {
-				ddLog.Warn("File does not exist, creating new one: ", err)
-			} else {
-				ddLog.Warn("Error deleting file: ", err)
-			}
-		} else {
-			ddLog.Info("Successfully deleted oversize log file, creating new one.")
+			return nil
+		}
+
+		if info.IsDir() {
+			return nil
 		}
 
-		file, err := os.Create(logFilePath)
+		fileInfo := &FileInfo{
+			filename:     info.Name(),
+			size:         info.Size(),
+			lastModified: info.ModTime(),
+		}
+
+		integrationID := fileNameToID(fileInfo.filename)
+
+		s.integrationToFile[integrationID] = fileInfo
+		s.combinedUsageSize += info.Size()
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
```

Comment:
That was actually what I had initially, but there is a case where it is possible when scanning the files that the actual oldest file doesn't get deleted if it hasn't been read yet. This always guarantee correct order of file deletion. Also, exceeding `combinedUsageMax` temporarily is okay since the files are already written to disk.